src/main/scala/com/scylladb/migrator/readers/Cassandra.scala

```diff
@@ -126,7 +128,17 @@ object Cassandra {
 
       timestampsToFields
         .map {
-          case ((ttl, writetime), fields) =>
+          case ((ttl, writetime), fields) => {
+            val writetimestamp = writetime.getOrElse(CassandraOption.Unset)
```

Comment:
This looks suspicious because `writetime` has type `Option[Long]` and `CassandraOption.Unset` has a type that is disjoint from `Long`. See my comment below for a suggestion.